### PR TITLE
Administrators management in admin area

### PR DESCRIPTION
**Pull request by bertocq**
This PR implements https://github.com/consul/consul/issues/1501

I've just reused existing similar feature for Moderators since it seemed to be equal as requested for Administrators and inventing myself a new way to manage them didn't seem reasonable. 

Anyway I would improve this by allowing to search also by name like on the Officials manage panel. I tried to add that feature but ended up messing too much. I'm still new to this codebase and not sure what's new way and what's the old. If you consider it a nice improvement I'll give it another try.

### CODE REVIEW

File: app/controllers/admin/administrators_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def create
+    @administrator.user_id = params[:user_id]
+    @administrator.save
+
+    redirect_to admin_administrators_path
+  end
+
+  def destroy
```
Comment (bertocq) [inline app/controllers/admin/administrators_controller.rb]: - Should a admin be able to remove himself from the Administrator list?
- Even in the scenario where he's the only admin in the list?

I guess no on both, but could only be no for the second one...

**Comment by bertocq** [comment]
As stated at https://github.com/consul/consul/pull/1562 I think broken tests (or at least most of them) was due to the time where the build was done

File: app/controllers/admin/administrators_controller.rb (hunk 1 above)
Comment (bertocq) [inline app/controllers/admin/administrators_controller.rb]: @decabeza Should I go on assuming both actions should be restricted? I was trying not to assume complete specs based on my opinion.

Comment (decabeza) [inline app/controllers/admin/administrators_controller.rb]: @bertocq both sounds perfect, you can restrict them!👌

Comment (bertocq) [inline app/controllers/admin/administrators_controller.rb]: Okey added that scenario to the spec https://github.com/consul/consul/pull/1560/commits/e9ea72113b30968e46225315e5e42ac97caea46b and simple current user check on the controller https://github.com/consul/consul/pull/1560/commits/fe5ddd019bc66fc95dffa8233b31b4552bcdf5a6

Now if the current user tries to remove himself from the administrators list he will get:
screen shot 2017-05-25 at 11 31 08

Any more details that could be polished? :)

**Comment by bertocq** [comment]
Hum that [failing spec](https://travis-ci.org/consul/consul/jobs/235952055) seems like a flaky test to me :(. Created a separate PR https://github.com/consul/consul/pull/1566 to fix it and avoid polluting this one. If you agree that may be the cause... can someone retry this branch travis build? :(

File: app/controllers/admin/administrators_controller.rb (hunk 1 above)
Comment (decabeza) [inline app/controllers/admin/administrators_controller.rb]: @bertocq it looks perfect! :wink:

**Comment by xuanxu** [comment]
🎉

**Pull request merged**